cmd/security-file-token-provider/main.go

```diff
@@ -0,0 +1,134 @@
+//
+// Copyright (c) 2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions and limitations under
+// the License.
+//
+// SPDX-License-Identifier: Apache-2.0'
+//
+
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"os"
+	"sync"
+	"syscall"
+
+	"github.com/edgexfoundry/edgex-go"
+	"github.com/edgexfoundry/edgex-go/internal"
+	"github.com/edgexfoundry/edgex-go/internal/pkg/bootstrap"
+	"github.com/edgexfoundry/edgex-go/internal/pkg/bootstrap/handlers"
+	bootstrapinterface "github.com/edgexfoundry/edgex-go/internal/pkg/bootstrap/interfaces"
+	"github.com/edgexfoundry/edgex-go/internal/pkg/bootstrap/startup"
+	"github.com/edgexfoundry/edgex-go/internal/pkg/usage"
+	"github.com/edgexfoundry/go-mod-registry/registry"
+
+	"github.com/edgexfoundry/edgex-go/internal/security/authtokenreader"
+	"github.com/edgexfoundry/edgex-go/internal/security/fileioperformer"
+	"github.com/edgexfoundry/edgex-go/internal/security/fileprovider"
+	"github.com/edgexfoundry/edgex-go/internal/security/tokenconfig"
+	"github.com/edgexfoundry/edgex-go/internal/security/vaultclient"
+	"github.com/edgexfoundry/go-mod-core-contracts/clients"
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/logger"
+)
+
+// Constants
+
+// Dependencies
+
+var fileProvider fileprovider.TokenProvider
+
+// Flag variables
+
+var configDir string
+var profileDir string
+var useRegistry bool
+
+// define and register command line flags
+func init() {
+	flag.BoolVar(&useRegistry, "registry", false, "Indicates the service should use Registry.")
+	flag.BoolVar(&useRegistry, "r", false, "Indicates the service should use Registry.")
+	flag.StringVar(&profileDir, "profile", "", "Specify a profile other than default.")
+	flag.StringVar(&profileDir, "p", "", "Specify a profile other than default.")
+	flag.StringVar(&configDir, "confdir", "", "Specify local configuration directory")
+}
+
+func main() {
+	startupTimer := startup.NewStartUpTimer(1, internal.BootTimeoutDefault)
+
+	flag.Usage = usage.HelpCallback
+	flag.Parse()
+
+	bootstrap.Run(
+		configDir,
+		profileDir,
+		internal.ConfigFileName,
+		useRegistry,
+		clients.SupportLoggingServiceKey,
+		fileprovider.Configuration,
+		startupTimer,
+		[]bootstrapinterface.BootstrapHandler{
+			handlers.NewStartMessage(clients.SupportLoggingServiceKey, edgex.Version).Handler,
```

Comment:
The NewStartMessage handler isn't necessary for your single-run non-persistent utility application.